src/dstack/_internal/core/backends/gcp/compute.py

```diff
@@ -87,9 +92,24 @@ def terminate_instance(
             backend_data_dict = json.loads(backend_data)
             zone = backend_data_dict["zone"]
         try:
-            self.instances_client.delete(
-                project=self.config.project_id, zone=zone, instance=instance_id
-            )
+            if "tpu" in instance_id:
```

Comment:
If my instance name has "tpu" in the name but it's not tpu, it will fail. You can store something in backend_data to indicate that the instance is TPU.